server/blob/blob-api/src/main/java/org/apache/james/blob/api/Store.java

```diff
@@ -110,4 +110,23 @@ public Impl(BlobPartsId.Factory<I> idFactory, Encoder<T> encoder, Decoder<T> dec
                 .map(decoder::decode);
         }
     }
+
+    class FixedLengthInputStream {
+
+        private final InputStream inputStream;
+        private final long contentLength;
+
+        public FixedLengthInputStream(InputStream inputStream, long contentLength) {
```

Comment:
you should add tests about this length

Reply:
it's flagged as resolved but I can't see the preconditions on this `long` value

Reply:
It's in this commit: b68b8bc82f51b161f5cb40a40a374719be7e6248

Reply:
it's not

Reply:
Sorry I had misunderstand your comment, this time it should be done.